File: ruby/100-Same-Tree.rb
def is_same_tree(p, q)
  if p.nil? && q.nil?
    true
  elsif p && q
    (p.val == q.val) &&
      is_same_tree(p.left, q.left) &&
      is_same_tree(p.right, q.right)
  else
    false
  end
end

File: go/BinaryTree/100_Same_Tree/100_Same_Tree.go
package sametree

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p == nil && q == nil {
		return true
	}
	if (q == nil) || (p == nil) {
		return false
	} else if p.Val != q.Val {
		return false
	}

	return isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)

}
